src/services/pcn-router/src/Router.cpp

```diff
@@ -775,6 +775,75 @@ void Router::addPorts(const std::string &name, const PortsJsonObject &conf) {
   add_port<PortsJsonObject>(name, conf);
   if (get_shadow())
     getPorts(name)->setMac(getPorts(name)->getMac());
+
+  // Creates function objects containing the callback functions
+  // to be called when there are notifications on one netdev
+  std::function<void(const std::string, const std::string)> f_ip = std::bind(&Router::iface_ip_notification,
```

Comment:
I think you can use a lambda function here. They are good in these cases because this avoid to create a new method that will make the code difficult to read.